[BUG]: TestSetLogger/TestRun failing with data race

### What happened?

One of the runs in the GitHub Actions failed with the following log indicating data race.

### Relevant log output

```Text
==================
WARNING: DATA RACE
Read at 0x00000128de40 by goroutine 11:
  github.com/doomshrine/gocosi.(*Driver).Run()
      /home/runner/work/gocosi/gocosi/gocosi.go:98 +0x51b
  github.com/doomshrine/gocosi.TestRun.func2()
      /home/runner/work/gocosi/gocosi/gocosi_test.go:102 +0xf6
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1629 +0x47

Previous write at 0x00000128de40 by goroutine 8:
  github.com/doomshrine/gocosi.SetLogger()
      /home/runner/work/gocosi/gocosi/gocosi.go:74 +0xca
  github.com/doomshrine/gocosi.TestSetLogger()
      /home/runner/work/gocosi/gocosi/gocosi_test.go:36 +0x74
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1629 +0x47

Goroutine 11 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1629 +0x805
  github.com/doomshrine/gocosi.TestRun()
      /home/runner/work/gocosi/gocosi/gocosi_test.go:94 +0x97
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1629 +0x47

Goroutine 8 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1576 +0x216
  testing.runTests()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:2034 +0x87c
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.20.7/x64/src/testing/testing.go:1906 +0xb44
  main.main()
      _testmain.go:84 +0x2fc
==================
--- FAIL: TestRun (0.00s)
    --- FAIL: TestRun/run_with_defaults (0.00s)
        testing.go:1446: race detected during execution of test
--- FAIL: TestSetLogger (0.00s)
    testing.go:1446: race detected during execution of test
FAIL
```

--- a/gocosi_test.go
+++ b/gocosi_test.go
@@ -29,9 +29,8 @@ import (
 	cosi "sigs.k8s.io/container-object-storage-interface-spec"
 )
 
+//nolint:paralleltest
 func TestSetLogger(t *testing.T) {
-	t.Parallel()
-
 	impl := logr.New(logr.Discard().GetSink())
 	SetLogger(impl)
 	assert.Equal(t, impl, log)

